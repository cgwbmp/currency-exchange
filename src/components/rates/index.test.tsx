import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Rates from '.';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render((
    <Rates
      currencies={['RUB', 'USD', 'EUR']}
      activeCurrency="RUB"
      rates={[
        { currency: 'USD', rate: 10 },
        { currency: 'EUR', rate: 9 },
      ]}
    />
  ), div);
  ReactDOM.unmountComponentAtNode(div);
});

it('DOM structure not changed', () => {
  const component = renderer.create((
    <Rates
      currencies={['RUB', 'USD', 'EUR']}
      activeCurrency="RUB"
      rates={[
        { currency: 'USD', rate: 10 },
        { currency: 'EUR', rate: 9 },
      ]}
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
