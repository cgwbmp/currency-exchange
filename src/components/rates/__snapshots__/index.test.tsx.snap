// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM structure not changed 1`] = `
<div
  className="ant-result ant-result-info"
>
  <div
    className="ant-result-icon"
  >
    <i
      aria-label="icon: fund"
      className="anticon anticon-fund"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="fund"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M926 164H94c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V196c0-17.7-14.3-32-32-32zm-40 632H134V236h752v560zm-658.9-82.3c3.1 3.1 8.2 3.1 11.3 0l172.5-172.5 114.4 114.5c3.1 3.1 8.2 3.1 11.3 0l297-297.2c3.1-3.1 3.1-8.2 0-11.3l-36.8-36.8a8.03 8.03 0 0 0-11.3 0L531 565 416.6 450.5a8.03 8.03 0 0 0-11.3 0l-214.9 215a8.03 8.03 0 0 0 0 11.3l36.7 36.9z"
        />
      </svg>
    </i>
  </div>
  <div
    className="ant-result-title"
  >
    <span
      className="rates-title"
    >
      Currency Rates to
      <div
        className="ant-select ant-select-enabled"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseOut={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
      >
        <div
          aria-autocomplete="list"
          aria-controls="test-uuid"
          aria-expanded={false}
          aria-haspopup="true"
          className="ant-select-selection
            ant-select-selection--single"
          onKeyDown={[Function]}
          role="combobox"
          tabIndex={0}
        >
          <div
            className="ant-select-selection__rendered"
          >
            <div
              className="ant-select-selection-selected-value"
              style={
                Object {
                  "display": "block",
                  "opacity": 1,
                }
              }
              title="RUB"
            >
              RUB
            </div>
          </div>
          <span
            className="ant-select-arrow"
            onClick={[Function]}
            style={
              Object {
                "WebkitUserSelect": "none",
                "userSelect": "none",
              }
            }
            unselectable="on"
          >
            <i
              aria-label="icon: down"
              className="anticon anticon-down ant-select-arrow-icon"
            >
              <svg
                aria-hidden="true"
                className=""
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </i>
          </span>
        </div>
      </div>
    </span>
  </div>
  <div
    className="ant-result-extra"
  >
    <div
      className="rates"
    >
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-12"
          style={Object {}}
        >
          <div
            className="ant-card ant-card-bordered"
          >
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="ant-statistic"
              >
                <div
                  className="ant-statistic-title"
                >
                  1 USD =
                </div>
                <div
                  className="ant-statistic-content"
                >
                  <span
                    className="ant-statistic-content-value"
                  >
                    <span
                      className="ant-statistic-content-value-int"
                    >
                      
                      10
                    </span>
                    <span
                      className="ant-statistic-content-value-decimal"
                    >
                      .00
                    </span>
                  </span>
                  <span
                    className="ant-statistic-content-suffix"
                  >
                    RUB
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col ant-col-12"
          style={Object {}}
        >
          <div
            className="ant-card ant-card-bordered"
          >
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="ant-statistic"
              >
                <div
                  className="ant-statistic-title"
                >
                  1 EUR =
                </div>
                <div
                  className="ant-statistic-content"
                >
                  <span
                    className="ant-statistic-content-value"
                  >
                    <span
                      className="ant-statistic-content-value-int"
                    >
                      
                      9
                    </span>
                    <span
                      className="ant-statistic-content-value-decimal"
                    >
                      .00
                    </span>
                  </span>
                  <span
                    className="ant-statistic-content-suffix"
                  >
                    RUB
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
